src/RegisterMapping.cpp

```diff
@@ -204,17 +204,17 @@ const std::vector<int16_t>& RegisterMapping::getAddresses() const
     return m_addresses;
 }
 
-int8_t RegisterMapping::getSlaveAddress() const
+int16_t RegisterMapping::getSlaveAddress() const
 {
     return m_slaveAddress;
 }
 
-void RegisterMapping::setSlaveAddress(int8_t slaveAddress)
+void RegisterMapping::setSlaveAddress(int16_t slaveAddress)
 {
     m_slaveAddress = slaveAddress;
 }
 
-int16_t RegisterMapping::getStartingAddress() const
+int32_t RegisterMapping::getStartingAddress() const
```

Comment:
Why this is double in size?

Reply:
Because, if you enter your address 35003 for example, it will become -30533.

Reply:
ok this is reg add, not slave add